pkg/controller/subscription/rhmiConfigs/main.go

```diff
@@ -72,53 +72,6 @@ func CreateInstallPlan(ctx context.Context, rhmiSubscription *olmv1alpha1.Subscr
 	return nil
 }
 
-func UpdateStatus(ctx context.Context, client k8sclient.Client, config *integreatlyv1alpha1.RHMIConfig, installplan *olmv1alpha1.InstallPlan, targetVersion string) error {
-	// Calculate the next maintenance window based on the maintenance schedule
-	if config.Spec.Maintenance.ApplyFrom != "" {
-		mtStart, _, err := getWeeklyWindowFromNow(config.Spec.Maintenance.ApplyFrom, time.Hour*WINDOW)
-		if err != nil {
-			return err
-		}
-
-		config.Status.Maintenance.ApplyFrom = mtStart.Format("2-1-2006 15:04")
-		config.Status.Maintenance.Duration = strconv.Itoa(WINDOW) + "hrs"
-	}
-
-	config.Status.TargetVersion = targetVersion
-
-	// If the install plan was approved, simply clear the status
-	if installplan.Spec.Approved {
-		config.Status.Upgrade.Scheduled.For = ""
-
-		return client.Status().Update(ctx, config)
-	}
-
-	// Calculate the upgrade schedule based on the spec:
-	// We can assume there's no error parsing the value as it was validated
-	notBeforeDays := *config.Spec.Upgrade.NotBeforeDays
-	waitForMaintenance := *config.Spec.Upgrade.WaitForMaintenance
-
-	upgradeSchedule := installplan.ObjectMeta.CreationTimestamp.Time.
-		Add(daysDuration(notBeforeDays)).UTC()
-
-	if waitForMaintenance {
-		var err error
-		upgradeSchedule, _, err = getWeeklyWindow(upgradeSchedule, config.Spec.Maintenance.ApplyFrom, time.Hour*WINDOW)
-		if err != nil {
-			return err
-		}
-	}
-
-	// Update the upgrade status
-	config.Status.Upgrade = integreatlyv1alpha1.RHMIConfigStatusUpgrade{
-		Scheduled: &integreatlyv1alpha1.UpgradeSchedule{
-			For: upgradeSchedule.Format(integreatlyv1alpha1.DateFormat),
-		},
-	}
-
-	return client.Status().Update(ctx, config)
-}
-
 func CanUpgradeNow(config *integreatlyv1alpha1.RHMIConfig, installation *integreatlyv1alpha1.RHMI) (bool, error) {
```

Comment:
I noticed there's a small possibility of a panic if the UpgradeAvailable field is set and this function is called before the RHMIConfig controller has calculated the schedule. I'd add this condition to return false if this is the case

```go
// If the upgrade schedule hasn't been calculated
if config.Status.Upgrade.Scheduled == nil {
	return false, nil
}
```